server/src/main/java/org/elasticsearch/index/fielddata/DenseVectorNumericByteValues.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.index.fielddata;
+
+import java.io.IOException;
+
+/**
+ * Use a simple array to store the values.
+ */
+public abstract class DenseVectorNumericByteValues {
```

Comment:
Do we need the 2 abstract classes of `DenseVectorNumericByteValues` and `DenseVectorNumericFloatValues` ? Can we make implementation more concise and just introduce concrete classed in VectorDVLeafFieldData?

Reply:
You are right, the two abstract classes were not necessary. I have made the implementation more concise by introducing concrete classes directly in `VectorDVLeafFieldData`.